src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
-}
+  const sourceArray = sourceString.split(';');
+
+  const objectProperties = sourceArray.reduce((acc, property) => {
+    const propertyArray = property.trim().split(':', 2);
+    const [keyName, keyValue] = propertyArray;
+
+    if (keyValue) {
```

Comment:
Here the previous comment talked about checking for undefined, but here it's better to check if both ``keyName`` and ``keyValue`` exist, this will prevent adding properties to the object with empty keys.

```suggestion
    if (keyName && keyValue) {
```